orc8r/cloud/go/services/tenants/obsidian/handlers/handlers.go

```diff
@@ -148,3 +159,55 @@ func DeleteTenantHandler(c echo.Context) error {
 	}
 	return c.NoContent(http.StatusNoContent)
 }
+
+func GetControlProxyHandler(c echo.Context) error {
+	tenantID, terr := obsidian.GetTenantID(c)
+	if terr != nil {
+		return terr
+	}
+
+	_, err := tenants.GetTenant(c.Request().Context(), tenantID)
+	switch {
+	case err == errors.ErrNotFound:
+		return echo.NewHTTPError(http.StatusNotFound, fmt.Errorf("Tenant %d does not exist", tenantID))
+	case err != nil:
+		return echo.NewHTTPError(http.StatusInternalServerError, err)
+	}
```

Comment:
Same here -- let's pull switch into a helper fn

Reply:
I did it, but I'm not sure it's an improvement since it's provides less clarity on exactly how the error handling is + not much less code

Reply:
I could also theoretically pass in a map of `StatusInternalServerError` to `error`, but I don't think that really decreases the code since that's the same thing as a `switch case`

Reply:
Mapping errors like this from a func is preferable imo. It 1. encapsulates the repeated logic in a single location, 2. yes it hides the impl, but you can jump to defn once and then you understand what it's doing, and 3. it strongly improves the skim-ability of the code